src/main/java/uk/nhs/hee/tis/trainee/forms/service/FormRelocateService.java

```diff
@@ -0,0 +1,182 @@
+import uk.nhs.hee.tis.trainee.forms.model.FormRPartA;
+import uk.nhs.hee.tis.trainee.forms.model.FormRPartB;
+import uk.nhs.hee.tis.trainee.forms.repository.FormRPartARepository;
+import uk.nhs.hee.tis.trainee.forms.repository.FormRPartBRepository;
+import uk.nhs.hee.tis.trainee.forms.service.exception.ApplicationException;
+
+
+@Slf4j
+@Service
+@Transactional
+@XRayEnabled
+public class FormRelocateService {
+
+  private final FormRPartARepository formRPartARepository;
+  private final FormRPartBRepository formRPartBRepository;
+  protected final AmazonS3 amazonS3;
+
+  @Value("${application.file-store.bucket}")
+  private String bucketName;
+
+  /**
+   * Constructor for Form Relocate service.
+   *
+   * @param formRPartARepository  spring data repository (Form R Part A)
+   * @param formRPartBRepository  spring data repository (Form R Part B)
+   * @param amazonS3              client for S3 service
+   */
+  public FormRelocateService(FormRPartARepository formRPartARepository,
+                             FormRPartBRepository formRPartBRepository,
+                             AmazonS3 amazonS3,
+                             String bucketName) {
+    this.formRPartARepository = formRPartARepository;
+    this.formRPartBRepository = formRPartBRepository;
+    this.amazonS3 = amazonS3;
+    this.bucketName = bucketName;
+  }
+
+  /**
+   * Relocate Form R.
+   */
+  public void relocateFormR(String formId, String targetTrainee) throws IOException {
+
+    // Get Form R from MongoDB by FormId
+    AbstractForm formR = getMoveFormInfoInDb(formId);
+
+    if (formR == null) {
+      log.error("Cannot find FormR with ID " + formId + " from DB.");
+      throw new ApplicationException("Cannot find FormR with ID " + formId + " from DB.");
+    }
+    else {
+      String formType = formR.getFormType();
+      String sourceTrainee = formR.getTraineeTisId();
+      try {
+        updateTargetTraineeInDb(formR, targetTrainee);
+        if (formR.getLifecycleState() != LifecycleState.DRAFT) {
+          relocateFormInS3(formType, formId, sourceTrainee, targetTrainee);
+        }
+      } catch (Exception e) {
+        log.error("Fail to relocate FormR to target trainee: " + e + ". Rolling back...");
+        try {
+          updateTargetTraineeInDb(formR, sourceTrainee);
+          if (formR.getLifecycleState() != LifecycleState.DRAFT) {
+            relocateFormInS3(formType, formId, targetTrainee, sourceTrainee);
+          }
+        } catch (Exception ex) {
+          log.error("Fail to roll back: " + ex);
+        }
+        throw new ApplicationException("Fail to relocate FormR to target trainee: " + e.toString());
+      }
+    }
+  }
+
+  private AbstractForm getMoveFormInfoInDb(String formId) {
+    try {
+      Optional<FormRPartA> optionalFormRPartA =
+          formRPartARepository.findById(UUID.fromString(formId));
+      Optional<FormRPartB> optionalFormRPartB =
+          formRPartBRepository.findById(UUID.fromString(formId));
```

Comment:
Only call this if `optionalFormRPartA` is empty, otherwise it's just a wasted DB request.